htdocs/core/lib/pdf.lib.php

```diff
@@ -1191,7 +1191,8 @@ function pdf_writelinedesc(&$pdf, $object, $i, $outputlangs, $w, $h, $posx, $pos
 		$labelproductservice = preg_replace('/(<img[^>]*src=")([^"]*)(&amp;)([^"]*")/', '\1\2&\4', $labelproductservice, -1, $nbrep);
 
 		//var_dump($labelproductservice);exit;
-
+		$up =  pdf_getlineupexcltax($object, $i, $outputlangs);
+		if(!empty($up) && strlen($up) >= 10) $labelproductservice .= "<br/>"; //For price higher to 100 000 or having too much decimals (depending of currency)
```

Comment:
Why adding a new line on description for long prices ? What is relation between price and description.

Why not changing the font used to make the MultiCell with a smaller font if pdf_getlineupexcltax return a too long value ?

Reply:
the relation is that pdf only handle multi lines on desc, too long price can get on next lines, even if desc is one line long, but if you prefer i can try change font but it won't be very beautiful. (you can try with an amount of 1 000 000)

Reply:
I met the same trouble for an invoice of several millions of euros. So I will use this example to experiment several solution to find a pleasant one.